src/inverted-index.js

```diff
@@ -81,36 +84,50 @@ class InvertedIndex {
   }
 
   /**
-   * Search method
+   * searchIndex
+   * *
    * Searches key words from the files that has been uploaded
+   * *
    * @param {any} filenames
    * @param {any} terms
    * @returns {object} Documents
    */
   searchIndex(filenames, ...terms) {
+    if (filenames !== null) {
+      if (!this.validateFileNames(filenames)) {
+        return 'filename does not exist';
+      }
+    }
     filenames = filenames || Object.keys(this.indices);
     const result = {};
     const searchTerms = terms.flatten();
-
-    // eslint-disable-next-line prefer-const
-    for (let searchTerm of searchTerms) {
+    searchTerms.forEach((searchTerm) => {
       result[searchTerm] = {};
-      // eslint-disable-next-line prefer-const
-      for (let index of filenames) {
-        result[searchTerm][index] = this.search(index, searchTerm);
-      }
-    }
+      filenames.forEach((index) => {
```

Comment:
### Critical

Variable names actually matter, you using `index` when you actually mean `filename` is diabolical!